inject-kotlin-test/build.gradle

```diff
@@ -43,3 +43,20 @@ tasks.named("compileGroovy") {
     dependsOn tasks.getByPath('compileKotlin')
     classpath += files(compileKotlin.destinationDir)
 }
+
+tasks.named("test", Test) {
+    if(JavaVersion.current().majorVersion.toInteger() >= 17) {
+        jvmArgs(
+                '--add-opens', 'jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED',
```

Comment:
This whole block is a bit scary!

Reply:
I agree, but without it we get

```
java.lang.IllegalAccessError: class org.jetbrains.kotlin.kapt3.base.KaptContext (in unnamed module @0x7a4f0f29) cannot access class com.sun.tools.javac.util.Context (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.util to unnamed module @0x7a4f0f29
```

All the way down until you end up with the above list.  I *think* it's because we're effectively hand-cranking Kapt from Groovy, but I have no proof...